### DISCUSSION TITLE
Detect Vertical PLane

### DISCUSSION THREAD

**Discussion by SyedMoizoffical** [sceneview/sceneview#512] [Q&A] [answered]
place object on vertical surface  ,Issue is that in not detect vertical surface ,Horizontally work fine ,
    
       sceneView = findViewById<ARSceneView?>(R.id.sceneView).apply {
            planeRenderer.isEnabled = true
            configureSession { session, config ->
                config.depthMode = when (session.isDepthModeSupported(Config.DepthMode.AUTOMATIC)) {
                    true -> Config.DepthMode.AUTOMATIC
                    else -> Config.DepthMode.DISABLED
                }
                config.planeFindingMode=Config.PlaneFindingMode.VERTICAL
                config.instantPlacementMode = Config.InstantPlacementMode.DISABLED
                config.lightEstimationMode = Config.LightEstimationMode.ENVIRONMENTAL_HDR
            }
            onSessionUpdated = { _, frame ->
                if (anchorNode == null) {
                    frame.getUpdatedPlanes()
                        .firstOrNull { it.type == Plane.Type.VERTICAL }
                        ?.let { plane ->
                           addAnchorNode(plane.createAnchor(plane.centerPose))
                        }
                }
            }
            onTrackingFailureChanged = { reason ->
                this@MainActivity.trackingFailureReason = reason
            }
        }

**Comment by UmerHamza** [comment]
sessionConfiguration = { session, config ->
                config.depthMode =
                    when (session.isDepthModeSupported(Config.DepthMode.AUTOMATIC)) {
                        true -> Config.DepthMode.AUTOMATIC
                        else -> Config.DepthMode.DISABLED
                    }
                config.lightEstimationMode =
                    Config.LightEstimationMode.ENVIRONMENTAL_HDR

                config.planeFindingMode = Config.PlaneFindingMode.VERTICAL.    // add this for vertical surface
            },

**Comment by SyedMoizoffical** [comment]
Already added

**Comment by ridcore** [comment, marked as answer]
Have you tried Config.PlaneFindingMode.HORIZONTAL_AND_VERTICAL?

In the little time I've been developing with this library, I don't notice that the VERTICAL configuration was available.

**Comment by UmerHamza** [comment]
try this 

frame.getUpdatedPlanes()
                    .firstOrNull() ?.let { plane ->
                       addAnchorNode(plane.createAnchor(plane.centerPose))
                    }

**Comment by hlefe** [comment]
Hello, I provided a solution to this issue here: https://github.com/SceneView/sceneview-android/issues/54#issuecomment-2508181759
